Re-aquire device functionality is missing

Noticed that a test is failing because of:

`AssertionError: Timed out retrying after 20000ms: Expected to find element: "[data-test="@menu/switch-device/refresh-button"]", but never found it.`

Turns out the ability to [re-acquire a device](https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/components/suite/Preloader/SuiteLayout/NavigationBar/DeviceSelector/DeviceSelector.tsx#L177) had been lost when `DeviceSelector` was re-written. 

[Figma](https://www.figma.com/file/Y2E8dcdNYrSMGIVRjYFXKp/%5Bs2.0%5D-Components?type=design&node-id=125-22036&mode=dev)

--- a/packages/suite/src/components/suite/NotificationCard.tsx
+++ b/packages/suite/src/components/suite/NotificationCard.tsx
@@ -78,11 +78,10 @@ const Wrapper = styled.div<{ elevation: Elevation; variant: NotificationCardVari
     align-items: center;
     background: ${({ theme, variant }) => getBackgroundColor(variant, theme)};
     margin-bottom: ${spacingsPx.xs};
+    gap: ${spacingsPx.md};
 `;
 
 const IconWrapper = styled.div`
-    margin-right: ${spacingsPx.md};
-
     @media screen and (max-width: ${variables.SCREEN_SIZE.SM}) {
         display: none;
     }

--- a/packages/suite/src/components/suite/Preloader/SuiteLayout/DeviceSelector/DeviceSelector.tsx
+++ b/packages/suite/src/components/suite/Preloader/SuiteLayout/DeviceSelector/DeviceSelector.tsx
@@ -1,9 +1,9 @@
-import { useState, useEffect, useRef } from 'react';
+import { useState, useEffect, useRef, MouseEventHandler } from 'react';
 
 import styled, { css } from 'styled-components';
 
 import { Image, Icon, DeviceAnimation } from '@trezor/components';
-import { selectDevicesCount, selectDevice } from '@suite-common/wallet-core';
+import { selectDevicesCount, selectDevice, acquireDevice } from '@suite-common/wallet-core';
 import * as deviceUtils from '@suite-common/suite-utils';
 import type { Timeout } from '@trezor/type-utils';
 import { SHAKE } from 'src/support/suite/styles/animations';
@@ -155,6 +155,13 @@ export const DeviceSelector = () => {
             }),
         );
 
+    const handleRefreshClick: MouseEventHandler = e => {
+        e.stopPropagation();
+        if (deviceNeedsRefresh) {
+            dispatch(acquireDevice(selectedDevice));
+        }
+    };
+
     return (
         <Wrapper
             data-test="@menu/switch-device"
@@ -183,6 +190,7 @@ export const DeviceSelector = () => {
                     <DeviceDetail>
                         <DeviceLabel>{selectedDevice.label}</DeviceLabel>
                         <DeviceStatusText
+                            onRefreshClick={handleRefreshClick}
                             device={selectedDevice}
                             walletLabel={
                                 walletLabel === undefined || walletLabel.trim() === ''
@@ -191,9 +199,11 @@ export const DeviceSelector = () => {
                             }
                         />
                     </DeviceDetail>
-                    <CaretContainer>
-                        <Icon size={20} icon="CARET_CIRCLE_DOWN" />
-                    </CaretContainer>
+                    {selectedDevice.state && (
+                        <CaretContainer>
+                            <Icon size={20} icon="CARET_CIRCLE_DOWN" />
+                        </CaretContainer>
+                    )}
                 </>
             )}
         </Wrapper>

--- a/packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceStatusText.tsx
+++ b/packages/suite/src/views/suite/SwitchDevice/DeviceItem/DeviceStatusText.tsx
@@ -1,13 +1,22 @@
-import styled, { useTheme } from 'styled-components';
+import styled, { css, useTheme } from 'styled-components';
 import { TrezorDevice } from '@suite-common/suite-types';
+import * as deviceUtils from '@suite-common/suite-utils';
 import { Icon } from '@trezor/components';
 import { spacingsPx, typography } from '@trezor/theme';
-import React from 'react';
+import React, { MouseEventHandler } from 'react';
 import { Translation, WalletLabeling } from 'src/components/suite';
 
-const Container = styled.span<{ color: string }>`
+const Container = styled.span<{ color: string; isAction?: boolean }>`
     ${typography.label}
     color: ${({ color }) => color};
+
+    ${({ isAction }) =>
+        isAction &&
+        css`
+            :hover {
+                opacity: 0.8;
+            }
+        `}
 `;
 const TextRow = styled.div`
     display: flex;
@@ -15,11 +24,33 @@ const TextRow = styled.div`
     gap: ${spacingsPx.xxs};
 `;
 
-type DeviceStatusTextProps = { device: TrezorDevice; walletLabel?: string };
+type DeviceStatusTextProps = {
+    device: TrezorDevice;
+    onRefreshClick?: MouseEventHandler;
+    walletLabel?: string;
+};
 
-export const DeviceStatusText = ({ device, walletLabel }: DeviceStatusTextProps) => {
+export const DeviceStatusText = ({
+    device,
+    onRefreshClick,
+    walletLabel,
+}: DeviceStatusTextProps) => {
     const theme = useTheme();
     const { connected } = device;
+    const deviceStatus = deviceUtils.getStatus(device);
+    const needsAttention = deviceUtils.deviceNeedsAttention(deviceStatus);
+
+    if (connected && needsAttention && onRefreshClick) {
+        return (
+            <Container isAction onClick={onRefreshClick} color={theme.textAlertYellow}>
+                <TextRow>
+                    <Icon icon="REFRESH" size={12} color={theme.textAlertYellow} />
+                    <Translation id="TR_SOLVE_ISSUE" />
+                </TextRow>
+            </Container>
+        );
+    }
+
     return (
         <Container
             color={connected ? theme.textPrimaryDefault : theme.textSubdued}
@@ -29,7 +60,7 @@ export const DeviceStatusText = ({ device, walletLabel }: DeviceStatusTextProps)
                 <Icon
                     icon={connected ? 'LINK' : 'UNLINK'}
                     size={12}
-                    color={connected ? theme.iconPrimaryDefault : theme.iconPrimaryDefault}
+                    color={connected ? theme.textPrimaryDefault : theme.textSubdued}
                 />
                 {walletLabel ? (
                     <WalletLabeling device={device} />

